2-running-automation-testing.md

```diff
@@ -0,0 +1,160 @@
+
+![automation](assets/2-automation.png)
+
+3. On the left hand side, you can select your preferred language, as well as any other variables you would like to adjust, such as **App Type**, **Device Group**, and **Orientation**. Adjusting the settings on the left side will affect the desiredCaps, which you can find in the right side of the window. 
+
+![automation-settings](assets/2-automationsettings.png)
+
+## 2.2 Create new Jenkins project to execute the automation test
+
+#### How to setup Jenkins to run automation testing
+
+Jenkins is an open source automation server that provides hundreds of plugins to support building, deploying, and automating any project. 
+
+Download Jenkins from https://jenkins.io/ and make sure you have a Jenkins server available with a public domain. 
+
+**2.2.1 Write the automation test script**
+
+* How to write a simple test
+
+For examples of automation tests, go to https://github.com/kobiton/samples . 
+
+Choose a language for your test script, and decide whether you want to test on Android or iOS, and either do a web test or an app test. Make sure in the code you specify your Kobiton username, API key, and information under desiredCaps. 
+
+**2.2.2 Create Jenkins project**
+
+On the left side on the page, click on New Item to start a new Jenkins project. 
+
+![new-item](assets/2-new-item.png)
+
+Enter in a name for the project and select 'Freestyle project'. Hit 'OK'. 
+
+![project](assets/2-project.png)
+
+Go to Configure in your Jenkins project. Add a build step, execute shell. 
+
+![build-execute-shell](assets/2-build-execute-shell.png)
+
+Enter this in the command. Enter your own Kobiton  username and API key as well as the desired device and platform you wish to test with. The below example will execute the android-web-test script sample from Kobiton. 
+```
+#!/bin/bash
+[ ! -d "samples" ] && git clone https://github.com/kobiton/samples.git || echo "Samples already exist"
+cd samples/javascript
+npm install
+npm run android-web-test
+```
+
+This guide is executing the NodeJS test script from the Kobiton samples, so we will add the NodeJS Plugin on Jenkins to execute NodeJS script as a build step. To do this, go to 'Manage Jenkins' -> 'Manage Plugins' -> Available. Search for NodeJS and install it.
+
+![manage-plugins](assets/2-manageplugins.png)
+
+If you are using a different language, you may try to find a suitable plugin for your project if necessary.
+
+Under build environment, check the box that says 'Provide Node & npm bin/ folder to PATH'.
+
+![build-environment](assets/2-build-environment.png)
+
+**2.2.3 Trigger Jenkins project by pushing on GitHub**
+
+Try Build now and check Kobiton cloud devices to see if a test session was created. 
+
+**2.2.4 Get the automation session data through Kobiton REST API**
+
+
+- Update test result to session (https://api.kobiton.com/docs/#update-session-information)
+
+```
+PUT /sessions/{sessionId}
+``` 
+To make a request:
```

Comment:
To make a request:

1. Encode your credential in base 64 for HTTP Basic Authentication, you may use below command and note the text result

```
echo -n <your username>:<your api-key> | base64
```

2. Use encoded base64 from above in `curl` commands, like below

```
curl -X GET https://api.kobiton.com/v1/sessions \
  -H 'Authorization: Basic dGVzdHVzZXI6MTIzZWQtMTIzZmFjLTkxMzdkY2E='
  -H 'Accept: application/json'
```

And below are necessary Kobiton Rest API endpoints that you may need

1.  Get session info

...